materialize-databricks/driver.go

```diff
@@ -188,15 +188,24 @@ func newTransactor(
 	}
 	defer db.Close()
 
+	schemas := map[string]struct{}{cfg.SchemaName: {}}
 	for _, binding := range bindings {
 		if err = d.addBinding(ctx, binding, open.Range); err != nil {
 			return nil, fmt.Errorf("addBinding of %s: %w", binding.Path, err)
 		}
+
+		schemas[binding.Path[0]] = struct{}{}
 	}
 
-	// Create volume for storing staged files
-	if _, err := db.ExecContext(ctx, fmt.Sprintf("CREATE VOLUME IF NOT EXISTS `%s`.`%s`;", cfg.SchemaName, volumeName)); err != nil {
-		return nil, fmt.Errorf("Exec(CREATE VOLUME IF NOT EXISTS %s;): %w", volumeName, err)
+	// Create a volume for storing staged files for all schemas in the scope of the materialization.
+	for sch := range schemas {
+		log.WithFields(log.Fields{
+			"schema":     sch,
+			"volumeName": volumeName,
+		}).Debug("creating volume for schema if it doesn't already exist")
+		if _, err := db.ExecContext(ctx, fmt.Sprintf("CREATE VOLUME IF NOT EXISTS `%s`.`%s`;", sch, volumeName)); err != nil {
```

Comment:
Are we using these separate volumes now? I think we would need to tweak the staged files machinery to specify the volume to which we upload files, otherwise by default we upload to the volume that resides in the connection's default schema which is `cfg.SchemaName`.

Reply:
When I tested this, I found that unless volumes were created for the alternate schemas of the bindings, the materialization would fail because the staged file and queries were trying to use a volume specific to that alternate schema already, but it didn't exist.

[Here](https://github.com/estuary/connectors/blob/dd5b4a206360c8ac79783ed5f8f0a90e1cb56075/materialize-databricks/driver.go#L234) is where the staging file path is set, and it uses `target.Path[0]` which is the binding's schema. The staging files are then initialized [a little later](https://github.com/estuary/connectors/blob/dd5b4a206360c8ac79783ed5f8f0a90e1cb56075/materialize-databricks/driver.go#L244-L245), and [use that file path](https://github.com/estuary/connectors/blob/dd5b4a206360c8ac79783ed5f8f0a90e1cb56075/materialize-databricks/staged_file.go#L176) containing the binding's schema to upload files. That path is also used when building [load](https://github.com/estuary/connectors/blob/dd5b4a206360c8ac79783ed5f8f0a90e1cb56075/materialize-databricks/driver.go#L284-L291) and [store](https://github.com/estuary/connectors/blob/dd5b4a206360c8ac79783ed5f8f0a90e1cb56075/materialize-databricks/driver.go#L403-L419) queries.

So from what I found, this volume creation for each binding's schema was necessary for bindings with alternate schemas to work.